lib/download.bash

```diff
@@ -41,6 +41,27 @@ need_cmd() {
   fi
 }
 
+# This function retries the download - if it fails due to network issues.
+retry_download(){
+  local max_retries=5
+  local retry_delay=2
+
+  for (( i=1 ; i<=max_retries ; i++ ))
+  do
+     if [ "$3" = "curl" ]; then
+        curl -sSfL "$1" -o "$2" && return 0
```

Comment:
nit: use long form equivalents where possible so the intent is more immediately apparent.

```suggestion
        curl --silent --show-error --fail --location "$1" -o "$2" && return 0
```